input/Glfw/GlfwInputDevice.cpp

```diff
@@ -21,17 +22,32 @@ namespace NovelRT::Input::Glfw
             throw Exceptions::InitialisationFailureException("GLFW3 failed to initialise.", std::string(output));
         }
 
-        if (window == nullptr)
+        if (device == nullptr)
         {
             throw Exceptions::NullPointerException(
-                "Could not initialise GLFW input service - null pointer was provided for window.");
+                "Could not initialise GLFW input service - null pointer was provided for windowing device.");
         }
 
         _logger.logInfoLine("Initialising GLFW input service.");
         _availableKeys = std::map<std::string, NovelKey>();
         _mappedActions = std::vector<InputAction>();
-        _window = reinterpret_cast<GLFWwindow*>(window);
-
+        _inputBufferCount = DEFAULT_INPUT_BUFFER_COUNT;
+        _keyStates = std::vector<std::unordered_map<int32_t, KeyStateFrameChangeLog>>(_inputBufferCount);
+        _previousBufferIndex = 0;
+        _currentBufferIndex = 1;
+
+        auto properDevice = std::reinterpret_pointer_cast<NovelRT::Windowing::Glfw::GlfwWindowingDevice>(device);
+        _window = properDevice->GetRawGLFWwindowHandle();
+        properDevice->KeyboardButtonChanged +=
+            [this](auto eventArgs) { ProcessKeyInput(eventArgs.key, eventArgs.action); };
+        properDevice->MouseButtonClicked +=
+            [this](auto eventArgs) { ProcessKeyInput(eventArgs.key, eventArgs.action); };
+        properDevice->CursorMoved += [this](auto eventArgs) {
+            NovelRT::Maths::GeoVector2F nativePos = NovelRT::Maths::GeoVector2F(static_cast<float>(eventArgs.x), static_cast<float>(eventArgs.y));
+            ProcessCursorMovement(nativePos);
+        };
+
+#pragma region KeyMapping
```

Comment:
This terrifies me - why have we added this?

Reply:
Please clarify - GitHub not be showing what you mean by "why have we added this" lol

Reply:
- I need clarification on what this `#pragma` is for.

Reply:
As per DMs, we agreed to move this into its own method - it was mainly for separating the big ol' keymapping section between GLFW Keys and NovelKeys